src/tm-threads.c

```diff
@@ -174,7 +174,7 @@ static int TmThreadTimeoutLoop(ThreadVars *tv, TmSlot *s)
                 Packet *p = PacketDequeue(tv->stream_pq);
                 SCMutexUnlock(&tv->stream_pq->mutex_q);
                 if (likely(p)) {
-                    if ((r = TmThreadsSlotProcessPkt(tv, fw_slot, p) != TM_ECODE_OK)) {
+                    if ((r = TmThreadsSlotProcessPkt(tv, fw_slot, p)) != TM_ECODE_OK) {
```

Comment:
Thanks for this.
Could we have a separation of the assignment and the condition over two different lines ?